### PR TITLE
First quantized select and prepare

### PR DESCRIPTION
**Pull request by fdmalone**
Adds final "outer" select and prepare bloqs which piece together the necessary subroutines from the previous first quantization PRs (contains part of #454 which should be merged first). Fixes some errors in previous incarnations of Select (signatures were either lacking some registers, which only really became apparent at a higher level).

I wrapped the multiplexed swaps #478  in to make splits / joins less obvious when moving from (3,) bitsize-sized registers to a (3*bitsize) big register. When combined together the costs match those quoted % some small things I've noted before. The remaining tasks would be to implement several lower level primitives (e.g. several arithmetic primitives need implementations and correct adjoints #485), and some of the state preparation routines are relatively staightforward but need controlled hadamards #237. 

I uses the existing SelectOracle and PrepareOracle as a model but maybe we want to include elements of #300 aswell (splitting big registers e.g.)

### CODE REVIEW

File: qualtran/bloqs/chemistry/pbc/first_quantization/first_quanization_test.py [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS,
-#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#  See the License for the specific language governing permissions and
-#  limitations under the License.
-
-from qualtran.testing import execute_notebook
-
-
-def test_notebook():
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/pbc/first_quantization/first_quanization_test.py:18]: ?

File: qualtran/bloqs/chemistry/pbc/first_quantization/__init__.py [hunk 2]
```diff
@@ -79,7 +79,4 @@
 ignored.
 """
 
-from .prepare_t import PrepareTFirstQuantization
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/pbc/first_quantization/__init__.py:82]: What's the strategy here of what is re-exported vs not

**Review by mpharrigan — APPROVED** [review]

File: qualtran/bloqs/chemistry/pbc/first_quantization/first_quanization_test.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/chemistry/pbc/first_quantization/first_quanization_test.py:18]: I've moved the first_quantization.py to select_and_prepare.py (the notebook is tested there.)

File: qualtran/bloqs/chemistry/pbc/first_quantization/__init__.py (hunk 2 above)
Comment (fdmalone) [reply qualtran/bloqs/chemistry/pbc/first_quantization/__init__.py:82]: I think just exporting the big bloqs is sufficient as they're most likely to be interacted with.

**Pull request merged**